pkg/apiserver/iam/provider/rolesyncer/rolesyncer.go

```diff
@@ -0,0 +1,35 @@
+// Copyright © 2023 Cisco Systems, Inc. and its affiliates.
+// All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package rolesyncer
+
+import (
+	"fmt"
+
+	"github.com/openclarity/vmclarity/api/models"
+	"github.com/openclarity/vmclarity/pkg/apiserver/iam"
+	"github.com/openclarity/vmclarity/pkg/apiserver/iam/provider/rolesyncer/jwtrole"
+)
+
+// New creates a new iam.RoleSyncer.
+func New(kind models.IamRoleSyncer) (iam.RoleSyncer, error) {
```

Comment:
From having a look around at different identity providers (github, slack etc) and looking at the goth library, it seems like in the majority of cases there isn't a way to customise the scopes/claims inside the token that is returned by the identity provider.

It seems like it would be a better idea to have a users construct in our DB which contains the user's identities and roles something like:

```
{
   "id": "<uuid 4>",
   "displayName": "Blah Blah",
   "identities": [
       {
           "Provider": "Github",
           "UserID": "blahblah@gmail.com"
        }
   ],
   "roles": [
       "foo",
       "bah",
       "baz"
   ],
}
```

this can then be used for all identity providers. This is kind of moving a feature of Zitadel into our code, but unless we want to make Zitadel a hard requirement for OpenClarity then I think we need it.

To support true multi-tenancy and the "SaaS" like nature of things we'd also probably want to make the identity providers configurable from the frontend/API so that you don't need to decide all of that as part of the deployment.

On that note, I don't think it would be a bad idea to consider making Zitadel a required component of the stack. Right now we havn't added anything to do with redirects to external providers and the required callbacks, user management UI, users with multiple identities from different providers, users which belong to different tenants etc. Unless we want to build all that ourselves then it seems like deploying Zitadel in the stack makes sense.